panic: runtime error: index out of range [n] of n

Provider ver `3.29.0` through latest ver `3.30.8`:
Terraform ver: `1.14.3`

```
Stack trace from the terraform-provider-pagerduty_v3.30.8 plugin:

panic: runtime error: index out of range [5] with length 5

goroutine 278 [running]:
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.flattenIncidentWorkflowSteps(0x14000342000, {0x14000805a70, 0x5, 0x1035677e0?}, 0x0)
        github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:512 +0x69c
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.flattenIncidentWorkflow(0x1400073d500, 0x14000342000, 0x1, {0x14000805a70, 0x5, 0x5}, 0x0)
        github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:494 +0x15c
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.fetchIncidentWorkflow.func1()
        github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:467 +0x2b8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.RetryContext.func1()
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/wait.go:30 +0x4c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:113 +0x140
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext in goroutine 22
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:86 +0x1c8

Error: The terraform-provider-pagerduty_v3.30.8 plugin crashed!
```

**To Reproduce**:
1. Add a resource block for a `pagerduty_schedule` and include a `for_each` with a reference for a `local` collection of users (per the Pagerduty docs as a supported practice for multiple users to receive a pageout at the same time for a single service)
2. Run a `terraform plan`
3. Error surfaces


This was previously working in older versions of the provider and fails on a fresh environment.

I am experiencing the same errors. I don't kn ow if it's related to a for_each as our config is generated by CDKTF, but this has been occurring for two days now.

Provider versions 3.24 through 3.30

`Stack trace from the terraform-provider-pagerduty_v3.30.9 plugin:
panic: runtime error: index out of range [4] with length 4
goroutine 1903 [running]:
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.flattenIncidentWorkflowSteps(0xc00107f2d0, {0xc000e506a0, 0x4, 0x10a7f00?}, 0x0)
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:512 +0x7ad
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.flattenIncidentWorkflow(0xc000a4b900, 0xc00107f2d0, 0x1, {0xc000e506a0, 0x4, 0x4}, 0x0)
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:494 +0x16c
github.com/PagerDuty/terraform-provider-pagerduty/pagerduty.fetchIncidentWorkflow.func1()
	github.com/PagerDuty/terraform-provider-pagerduty/pagerduty/resource_pagerduty_incident_workflow.go:467 +0x319
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.RetryContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/wait.go:30 +0x4e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:113 +0x1a3
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext in goroutine 1901
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:86 +0x237`

--- a/pagerduty/resource_pagerduty_incident_workflow.go
+++ b/pagerduty/resource_pagerduty_incident_workflow.go
@@ -508,7 +508,7 @@ func flattenIncidentWorkflowSteps(iw *pagerduty.IncidentWorkflow, specifiedSteps
 
 		var inputNames []string
 		inlineInputs := make(map[string][]*SpecifiedStep)
-		if !isImport {
+		if !isImport && i < len(specifiedSteps) {
 			specifiedStep := *specifiedSteps[i]
 			inputNames = specifiedStep.SpecifiedInputNames
 			inlineInputs = specifiedStep.SpecifiedInlineInputs
@@ -575,7 +575,7 @@ func flattenIncidentWorkflowStepInlineStepsInputSteps(
 
 		var inputNames []string
 		inlineInputs := make(map[string][]*SpecifiedStep)
-		if !isImport {
+		if !isImport && i < len(specifiedSteps) {
 			specifiedStep := *specifiedSteps[i]
 			inputNames = specifiedStep.SpecifiedInputNames
 			inlineInputs = specifiedStep.SpecifiedInlineInputs

--- a/pagerduty/resource_pagerduty_incident_workflow_test.go
+++ b/pagerduty/resource_pagerduty_incident_workflow_test.go
@@ -664,3 +664,164 @@ func testAccPreCheckIncidentWorkflows(t *testing.T) {
 		t.Skip("PAGERDUTY_ACC_INCIDENT_WORKFLOWS not set. Skipping Incident Workflows-related test")
 	}
 }
+
+// TestFlattenIncidentWorkflowSteps_IndexOutOfRange tests the scenario where
+// the API returns more steps than what exists in specifiedSteps.
+// This can happen when a workflow is modified outside of Terraform.
+func TestFlattenIncidentWorkflowSteps_IndexOutOfRange(t *testing.T) {
+	// Create a workflow with 5 steps (simulating API response)
+	iw := &pagerduty.IncidentWorkflow{
+		ID:   "PTEST123",
+		Name: "Test Workflow",
+		Steps: []*pagerduty.IncidentWorkflowStep{
+			{
+				ID:   "step1",
+				Name: "Step 1",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action1",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+			{
+				ID:   "step2",
+				Name: "Step 2",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action2",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+			{
+				ID:   "step3",
+				Name: "Step 3",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action3",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+			{
+				ID:   "step4",
+				Name: "Step 4",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action4",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+			{
+				ID:   "step5",
+				Name: "Step 5",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action5",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+		},
+	}
+
+	// Create specifiedSteps with only 4 elements (less than iw.Steps)
+	// This simulates the scenario where Terraform state has 4 steps but API returns 5
+	specifiedSteps := []*SpecifiedStep{
+		{
+			SpecifiedInputNames:   []string{},
+			SpecifiedInlineInputs: make(map[string][]*SpecifiedStep),
+		},
+		{
+			SpecifiedInputNames:   []string{},
+			SpecifiedInlineInputs: make(map[string][]*SpecifiedStep),
+		},
+		{
+			SpecifiedInputNames:   []string{},
+			SpecifiedInlineInputs: make(map[string][]*SpecifiedStep),
+		},
+		{
+			SpecifiedInputNames:   []string{},
+			SpecifiedInlineInputs: make(map[string][]*SpecifiedStep),
+		},
+	}
+
+	// Call the function with isImport=false to trigger the code path that accesses specifiedSteps
+	// Without the fix, this will panic with: index out of range [4] with length 4
+	// With the fix, this should not panic
+	result := flattenIncidentWorkflowSteps(iw, specifiedSteps, false)
+
+	// Verify we got the expected number of steps back
+	if len(result) != 5 {
+		t.Errorf("Expected 5 steps in result, got %d", len(result))
+	}
+
+	// Verify all steps have the expected fields
+	for i, step := range result {
+		if step["id"] == nil {
+			t.Errorf("Step %d missing id field", i)
+		}
+		if step["name"] == nil {
+			t.Errorf("Step %d missing name field", i)
+		}
+		if step["action"] == nil {
+			t.Errorf("Step %d missing action field", i)
+		}
+	}
+}
+
+// TestFlattenIncidentWorkflowSteps_EmptySpecifiedSteps tests with empty specifiedSteps
+// which simulates a fresh import or read operation
+func TestFlattenIncidentWorkflowSteps_EmptySpecifiedSteps(t *testing.T) {
+	iw := &pagerduty.IncidentWorkflow{
+		ID:   "PTEST456",
+		Name: "Test Workflow 2",
+		Steps: []*pagerduty.IncidentWorkflowStep{
+			{
+				ID:   "step1",
+				Name: "Step 1",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action1",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+			{
+				ID:   "step2",
+				Name: "Step 2",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action2",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+		},
+	}
+
+	// Empty specifiedSteps
+	specifiedSteps := []*SpecifiedStep{}
+
+	// This should not panic
+	result := flattenIncidentWorkflowSteps(iw, specifiedSteps, false)
+
+	if len(result) != 2 {
+		t.Errorf("Expected 2 steps in result, got %d", len(result))
+	}
+}
+
+// TestFlattenIncidentWorkflowSteps_Import tests the import path
+func TestFlattenIncidentWorkflowSteps_Import(t *testing.T) {
+	iw := &pagerduty.IncidentWorkflow{
+		ID:   "PTEST789",
+		Name: "Test Workflow 3",
+		Steps: []*pagerduty.IncidentWorkflowStep{
+			{
+				ID:   "step1",
+				Name: "Step 1",
+				Configuration: &pagerduty.IncidentWorkflowActionConfiguration{
+					ActionID: "action1",
+					Inputs:   []*pagerduty.IncidentWorkflowActionInput{},
+				},
+			},
+		},
+	}
+
+	// When isImport=true, specifiedSteps is not accessed, so it can be empty
+	specifiedSteps := []*SpecifiedStep{}
+
+	result := flattenIncidentWorkflowSteps(iw, specifiedSteps, true)
+
+	if len(result) != 1 {
+		t.Errorf("Expected 1 step in result, got %d", len(result))
+	}
+}

